ie/node-id.go

```diff
@@ -46,6 +46,33 @@ func NewNodeID(ipv4, ipv6, fqdn string) *IE {
 	return New(NodeID, p)
 }
 
+// NewNodeIDHeuristic creates a new NodeID IE.
+func NewNodeIDHeuristic(nodeID string) *IE {
+	if nodeID == "" {
+		return nil
+	}
+
+	var p []byte
+	ip := net.ParseIP(nodeID)
+	if ip != nil {
+		if ip.To4() != nil { // IPv4
+			p = make([]byte, 5)
+			p[0] = NodeIDIPv4Address
+			copy(p[1:], ip.To4())
```

Comment:
```suggestion
		if v4 := ip.To4(); v4 != nil { // IPv4
			p = make([]byte, 5)
			p[0] = NodeIDIPv4Address
			copy(p[1:], v4)
```

You can avoid calling `To4()` twice (not sure if it really affect the performance - maybe compiler does something or maybe not :p)